### ISSUE TITLE
pets

### ISSUE THREAD

**Issue by tobyplowy** [D00Med/farlands#99]
i will expand later

**Comment by GreenXenith** [comment]
Flying banana.

**Comment by tobyplowy** [comment]
@GreenXenith ok thats weard but it can be somting for
https://github.com/D00Med/farlands/issues/97

**Comment by GreenXenith** [comment]
I was half-joking :P

**Comment by tobyplowy** [comment]
![evil](https://cloud.githubusercontent.com/assets/16853304/26414339/2cf1e456-40af-11e7-9ba4-78a27384c5a5.png)
[cubelet.zip](https://github.com/D00Med/farlands/files/1026276/cubelet.zip)

**Comment by GreenXenith** [comment]
Yes. Just yes.

**Comment by tobyplowy** [comment]
> I was half-joking :P

that doesent matter i need weard stuff for loot

**Comment by tobyplowy** [comment]
cubelet is inspierd by the Blocklings mod for mc

**Comment by tobyplowy** [comment]
i dout take credit for the idea

**Comment by D00Med** [comment]
@tobyplowy too bad, nice idea :P

**Comment by DS-Minetest** [comment]
> Flying banana.

Use banana as boomerang!

@tobyplowy I like that minion.
I think, we need a pet API.
Make the pets follow the player, maybe teleport them near player if they are too far away.
Don't make a way to make the pet stop at a position.
Use items to spawn pets.
Perhaps only one pet active once.
Give the player advantages of the pet.

**Comment by D00Med** [comment]
@DS-Minetest That's a good idea, mobs_redo isn't very good for pets

**Comment by tobyplowy** [comment]
@D00Med minetests new mob api maybe good hopfully they will work on it in the next unstable

**Comment by D00Med** [comment]
@tobyplowy yeah hopefully

**Comment by tobyplowy** [comment]
@D00Med well looks like the new mob ai is abandond :/

**Comment by D00Med** [comment]
@tobyplowy  That's a shame :/

**Comment by tobyplowy** [comment]
> I think, we need a pet API.
> Make the pets follow the player, maybe teleport them near player if they are too far away.
> Don't make a way to make the pet stop at a position.
> Use items to spawn pets.
> Perhaps only one pet active once.

@DS-Minetest wuld you bne interested to make the pet api for V2 or V3?

**Comment by tobyplowy** [comment]
[pug.zip](https://github.com/D00Med/farlands/files/1290912/pug.zip)

![pug](https://user-images.githubusercontent.com/16853304/30252708-ab4beab8-9677-11e7-98d7-c6d157db6ed0.png)

**Comment by DS-Minetest** [comment]
>wuld you bne interested to make the pet api for V2 or V3?

Nah, I'm rather inexperienced with mobs and I don't think I could make anything nice soon.

**Comment by tacigar** [comment]
@tobyplowy Should we discuss pet API here?

**Comment by tobyplowy** [comment]
@tacigar yes

**Comment by tobyplowy** [comment]
@tacigar I updated the top comment to show you my idea (keep in mind that it is just an idea)
I'm interested to hear what you think about it

**Comment by tacigar** [comment]
@tobyplowy 
Accessories and changeable skin are nice idea.
But, what should pets do?
Do you need functions such as farming, cutting trees, etc.?

**Comment by tobyplowy** [comment]
> But, what should pets do?

They can do all kinds of stuff like there can be a glow worm pet that gives light (usefull for if you're in caves)

Some pets can attack enemies

Maybe we can add a mimic chest pet that follows the player and you change use it as mobile storage
(Basically it's a walking chest)

And many more we have to come up with some  so if you have more ideas please let me know :)


> Do you need functions such as farming, cutting trees, etc.?

Not really but we do need that stuff for our vilagers (they are very stupid XD )
I can make an issue for that if you rather work on that

**Comment by tacigar** [comment]
> Pets shuld not have a bitbox (so they Don't Get in the way in small Spaces)
> You should also not be able to hurt them
Sorry, I cannot interpret it.
Does It meen that pets don't have collision box?

> Also a pet capsule holder wuld be nice (is basically a block and you can put your pet capsule in it
> And your pet will stay in the radius of that block) this is good to have your pet walking around in your house
Sorry, I can not imagine...

> Not really but we do need that stuff for our vilagers (they are very stupid XD )
> I can make an issue for that if you rather work on that
I will update villagers next time.
Now I just want to do a pet API :)

**Comment by tobyplowy** [comment]
> Sorry, I cannot interpret it.
> Does It meen that pets don't have collision box

Yes You should be able to walk through them as if they are ghosts (as if they are not there)

> Sorry, I can not imagine...

It's a block and you can put the pet capsule (pet spaun item) in it, 
the pet will stay close to the block so that the pet can walk around in the house

I hope I'm not using to many complicated words (also don't feel bad for not understanding)
I can emagine English to Japanese is hard to translate :)

**Comment by tobyplowy** [comment]
@tacigar maybe you should translate the inportant points in to Japanese so that you don't have to read my big comment Everytime :)

**Comment by tacigar** [comment]
@tobyplowy 
Thanks, I almost understood.

> I hope I'm not using to many complicated words 

Yes, thanks.
I can understand words, but I'm confused when I try to understand the meaning of long sentences :(
I will do my best to get to the meaning of the sentence quickly.

**Comment by tacigar** [comment]
> maybe you should translate the inportant points in to Japanese so that you don't have to read my big comment Everytime :)

Thanks for your advice :)

**Comment by tobyplowy** [comment]
> Thanks, I almost understood

Almost so that's not all, what do you not understand yet?
(I can explain it)

> Thanks for your advice :)

Haha no problem :)
I need to thank you

**Comment by tobyplowy** [comment]
@MarkuBu what bransh shuld tacigar work on is the 1.1 bransh good?

**Comment by DS-Minetest** [comment]
>Yes You should be able to walk through them as if they are ghosts (as if they are not there)

Better set an infinitely small selectionbox, a normal collitionbox, physical to `true` and collide_with_objects to `false`, else the pet will fall through floor.

**Comment by MarkuBu** [comment]
@tobyplowy I assume it is a mod and nothing is changed at MTG, so go ahead and add it to 1.1

**Comment by tobyplowy** [comment]
@tacigar I guess you have all the information you need any other questions?

**Comment by tacigar** [comment]
> Better set an infinitely small selectionbox, a normal collitionbox, physical to true and collide_with_objects to false, else the pet will fall through floor.

I see :)

@tobyplowy 
Thanks, I will begin to make API.
I will ask if something I do not understand comes out :)

**Comment by tobyplowy** [comment]
@tacigar can you make it for the 1.1 branch
https://github.com/D00Med/farlands/tree/1.1

**Comment by tacigar** [comment]
@tacigar OK

**Comment by tacigar** [comment]
Is directory name `mods/pets_farlands`, OK?

**Comment by tacigar** [comment]
or customize "mobs_redo" mod?

**Comment by tobyplowy** [comment]
>Is directory name mods/pets_farlands, OK?

Yes that's perfect :)
Sorry for the late reply (we have different time zones)

**Comment by tobyplowy** [comment]
> or customize "mobs_redo" mod?

No that's probably not a good idea :)

**Comment by tobyplowy** [comment]
>Is directory name mods/pets_farlands, OK?

Yes that's perfect :)
Sorry for the late reply (we have different time zones)

> or customize "mobs_redo" mod?

No that's probably not a good idea :)

**Comment by tobyplowy** [comment]
@tacigar we use discord to talk so if you are interested https://discord.gg/wEGywb

**Comment by tacigar** [comment]
Sorry, I have been late replying :(

OK I'll make a new directory.

**Comment by tacigar** [comment]
@tobyplowy 
Oh... is discord VoIP?
Sorry, I cannot speak English :(

**Comment by tobyplowy** [comment]
> Sorry, I have been late replying :(

dount worry :D

wat is VoIP?

> Sorry, I cannot speak English :(

what you are typing seems good to me

**Comment by tacigar** [comment]
VoIP stands for "Voice over IP".
It is a service that you can talk on the Internet like Skype.

> what you are typing seems good to me

I am getting a lot of help of Google translation :(

**Comment by tobyplowy** [comment]
@tacigar ah you can also talk with text on discord :)

> I am getting a lot of help of Google translation :(

Japanese to English must be very good than 
(I can't say the same for other languages especially Russian XD )

So what will be the first thing you add for the pet api

And do you have any suggestions for pets (so I can make the models for you)

**Comment by tobyplowy** [comment]
@D00Med @22i we need animation for the above models

**Comment by tacigar** [comment]
@tobyplowy 
Sorry, for the late reply.
OK, I join.

> And do you have any suggestions for pets (so I can make the models for you)

I will write it when my idea is settled :)

**Comment by tacigar** [comment]
Sorry, discode invitation is expired 😭

**Comment by tobyplowy** [comment]
@tacigar there you go a new one https://discord.gg/Sju8qS

**Comment by tacigar** [comment]
@tobyplowy 
Thanks!

**Comment by tobyplowy** [comment]
@tacigar can you please go in the general tab and say hi so your name shows up on discord?
